core/controller/src/main/scala/whisk/core/controller/Actions.scala

```diff
@@ -118,15 +118,19 @@ trait WhiskActionsApi extends WhiskCollectionAPI with PostActionActivation with
   protected override def innerRoutes(user: Identity, ns: EntityPath)(implicit transid: TransactionId) = {
     (entityPrefix & entityOps & requestMethod) { (segment, m) =>
       entityname(segment) { outername =>
-        pathEnd {
+        (pathEnd & put & entity(as[WhiskActionPut])) { (content) =>
```

Comment:
@asteed can you add a comment explaining why you're short-circuiting the main dispatcher?

Reply:
@rabbah the issue at hand was to be able to extract the request payload (specifically for the `kind` property) and marshall it into `entitlementProvider.checkResource` so that it can be used to determine the kind at the time of action creation. I'd appreciate any guidance if you know of any alternatives to this approach